src/main/resources/com/mathworks/ci/RunMatlabTestsBuilder/runMatlabTests.m

```diff
@@ -133,11 +131,18 @@
 results = runner.run(suite);
 failed = any([results.Failed]);
 
+function fileName = getFullFileForReport(filePath)
+[filepath,name,ext] = fileparts(filePath);
+if (filepath == "")
```

Comment:
This will not work on versions of MATLAB that don't support strings. I am not sure why this is necessary anyway. It doesn't seem right to me that we deconstruct the file parts, sometimes mkdir if needed, and then put the file back together and return the same string (except when we are in pwd and we then move to an absolute path instead of a relative one, which is inconsistent with other relative paths. Also, it is named getFullfileForReport but it accepts the full file for the report and returns the full file for the report but has side effects.

Rather than doing this, can we simply call mkdirIfNeeded with the file path? Could either be:

`function mkdirIfNeeded(dir)
dirExists = isempty(dir) || exist(dir,'dir') == 7;
if ~dirExists		
    mkdir(dir);		    
end	`

where we call it like so: `mkdirIfNeeded(fileparts(pdfReportPath))`

or we just pass in the whole path each time and we issue the file parts in mkdirIfNeeded:

`function preparePath(path)
dir = fileparts(path);
dirExists = isempty(dir) || exist(dir,'dir') == 7;
if ~dirExists		
    mkdir(dir);		    
end`

Note maybe preparePAth is a better name for it at that point, and that is a name that implies possible side effects (like creating a dir)

Reply:
Done. implemented preparePath()